engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/DefaultOAuth2ServiceFactory.scala

```diff
@@ -19,36 +23,65 @@ class DefaultOAuth2Service(clientApi: OAuth2ClientApi[DefaultProfileResponse, De
     }
   }
 
-  override def authorize(token: String): Future[LoggedUser] = {
-    clientApi.profileRequest(token).map { profile =>
-      val userRoles = DefaultOAuth2ServiceFactory.getUserRoles(profile.email, configuration)
-      val rulesSet = RulesSet.getOnlyMatchingRules(userRoles, configuration.rules, allCategories)
-      val username = profile.login.getOrElse(profile.id.toString)
-      LoggedUser(id = profile.id.toString, username = username, rulesSet=rulesSet)
-    }
-  }
+  def authorize(token: String): Future[LoggedUser] = clientApi.profileRequest(token).map(getLoggedUser(_, configuration, allCategories))
 }
 
-class DefaultOAuth2ServiceFactory extends OAuth2ServiceFactory {
-  override def create(configuration: OAuth2Configuration, allCategories: List[String]): OAuth2Service =
-    DefaultOAuth2ServiceFactory.defaultService(configuration, allCategories)
-}
+class DefaultOAuth2ServiceFactoryWithProfileFormat[ProfileResponse: Decoder](implicit getLoggedUser: GetLoggedUserType[ProfileResponse]) {
+  def defaultService(configuration: OAuth2Configuration, allCategories: List[String]): OAuth2Service =
+    new DefaultOAuth2Service[ProfileResponse](OAuth2ClientApi[ProfileResponse, DefaultAccessTokenResponse](configuration), configuration, allCategories)
 
-object DefaultOAuth2ServiceFactory extends {
-  val defaultUserRole = "User"
+  def service(configuration: OAuth2Configuration, allCategories: List[String])(implicit backend: SttpBackend[Future, Nothing, NothingT], ec: ExecutionContext): OAuth2Service =
+    new DefaultOAuth2Service[ProfileResponse](new OAuth2ClientApi[ProfileResponse, DefaultAccessTokenResponse](configuration), configuration, allCategories)
+}
 
-  def apply(): DefaultOAuth2ServiceFactory = new DefaultOAuth2ServiceFactory()
+object DefaultOAuth2ServiceFactoryWithProfileFormat {
+  type GetLoggedUserType[ProfileResponse] = (ProfileResponse, OAuth2Configuration, List[String]) => LoggedUser
```

Comment:
Let's make a trait for that, it will be more handy to implement for extensions

Reply:
OK